Darwin/Services/StitchRemoteMongoDBService/StitchRemoteMongoDBServiceTests/SyncIntTests.swift

```diff
@@ -1788,19 +1793,21 @@ class SyncIntTests: BaseStitchIntTestCocoaTouch {
         XCTAssertEqual(expectedDocument, SyncIntTestUtilities.withoutSyncVersion(remoteColl.findOne(doc1Filter)!))
         XCTAssertNil(coll.findOne(doc1Filter))
 
-        // go online to begin the syncing process. A conflict should have
-        // occurred because both the local and remote instance of this document have no version
-        // information, meaning that the sync pass was forced to raise a conflict. our local
-        // delete should be synced to the remote, because we set up the conflict handler to
-        // have local always win. assert that this is reflected remotely and locally.
+        // go online to begin the syncing process. When doing R2L first, a version and
+        // apply operation should have occurred because both the local and remote instance of this
+        // document have no version. our local delete should then cause syncing to stop and
+        // the local document deleted.
+        // (As a historical note, when we used to permit L2R first, no conflict would be raised
+        // since we didn't get a chance to fetch stale documents, potentially resulting in the
+        // loss of events
         goOnline()
         // do one sync pass to get the local delete to happen via conflict resolution
         try ctx.streamAndSync()
         // do another sync pass to get the local delete resolution committed to the remote
         try ctx.streamAndSync()
 
         // make sure that a conflict was raised
```

Comment:
Because the remote version application (performed inline, not in L2R pass as it likely should have been in future all things being equal) creates a remote update event where the sync version field is changed. Because the instance ID and version match and there's no change to the non-version data, it's dropped as a stale event. The timing is weird because it's all happening in R2L instead of L2R